File: src/test/cpp/sorting/bubblesort/bubble_sort_test.cpp
//
// Created by nisha on 6/1/2020.
//

#include <gtest/gtest.h>

#include "main/cpp/sorting/bubblesort/bubble_sort.h"
#include "main/cpp/my_utils.h"

TEST(Sorting, BubbleSortTest) {
  std::vector<int> arr = my_utils::createRandVector(10000, 1, 1000000);
  std::vector<int> copy = arr;
  
  std::sort(copy.begin(), copy.end());
  bubble_sort(arr);
  ASSERT_EQ(copy, arr);
}

File: src/test/java/Sorting/BubbleSortTest.java
package Sorting;

import static org.junit.Assert.assertArrayEquals;

import org.junit.Before;
import org.junit.Test;
import sorting.bubblesort.BubbleSort;

public class BubbleSortTest {

  private int[] arr;

  @Before
  public void setup() {
    arr = new int[] {5, 3, 2, 1, 5};
  }

  @Test
  public void test() {
    BubbleSort.sort(arr);
    assertArrayEquals(new int[] {1, 2, 3, 5, 5}, arr);
  }
}
